src/viadot/orchestration/prefect/utils.py

```diff
@@ -28,6 +28,394 @@
+
+        Returns:
+            list: A list of dates in ascending order.
+        """
+        current_date = pendulum.now(self.dynamic_date_timezone)
+        result = [
+            current_date.subtract(days=i).format("YMMDD") for i in range(last_days)
+        ][
+            ::-1
+        ]  # Reversed to ascending order
+        return result
+
+    def process_first_days(
+        self, month_name: str, year: int, num_days: int
+    ) -> List[str]:
+        """
+        Generate a list of the first X days of a given month and year.
+
+        Args:
+            month_name (str): The name of the month.
+            year (str): The year.
+            num_days (int): The number of days to include.
+
+        Returns:
+            list: A list of dates for the first X days in ascending order.
+        """
+        start_date = pendulum.date(
+            int(year), pendulum.parse(month_name, strict=False).month, 1
+        )
+        result = [
+            start_date.add(days=i).format("YMMDD") for i in range(num_days)
+        ]  # Ascending order
+        return result
+
+    def process_last_days(self, month_name: str, year: int, num_days: int) -> List[str]:
+        """
+        Generate a list of the last X days of a given month and year.
+
+        Args:
+            month_name (str): The name of the month.
+            year (str): The year.
+            num_days (int): The number of days to include.
+
+        Returns:
+            list: A list of dates for the last X days in ascending order.
+        """
+        start_date = pendulum.date(
+            int(year), pendulum.parse(month_name, strict=False).month, 1
+        )
+        end_date = start_date.end_of("month")
+        result = [end_date.subtract(days=i).format("YMMDD") for i in range(num_days)][
+            ::-1
+        ]  # Reversed to ascending order
+        return result
+
+    def process_last_day_of_month(
+        self, year: str, month_name: str
+    ) -> pendulum.DateTime:
+        """
+        Retrieve the last day of a specified month and year.
+
+        Args:
+            year (str): The year.
+            month_name (str): The name of the month.
+
+        Returns:
+            pendulum.DateTime: A date object containing the last day of the specified month.
+        """
+        month_num = pendulum.parse(month_name, strict=False).month
+        date = pendulum.date(int(year), month_num, 1).end_of("month")
+        return date
+
+    def process_x_years_ago(self, year: str) -> str:
+        """
+        Retrieve the year of a date X years from now.
+
+        Args:
+            year (str): The year.
+
+        Returns:
+            str: A string containing the year of the specified time ago.
+        """
+        current_date = pendulum.now()
+        result = current_date.subtract(years=year).format("Y")
+        return result
+
+    def get_date_x_ago_full_date(self, number: int, unit: str) -> pendulum.DateTime:
```

Comment:
mark this function internal by adding `_`.
```suggestion
    def _get_date_x_ago_full_date(self, number: int, unit: str) -> pendulum.DateTime:
```